classes/package.php

```diff
@@ -209,34 +213,36 @@ public function store_in_db(int $contextid = 0) {
             "url" => $this->url,
             "icon" => $this->icon,
             "license" => $this->license
-        ];
-
-        $transaction = $DB->start_delegated_transaction();
-        $packageid = $DB->insert_record('qtype_questionpy_package', $packagedata);
-
-        // For each language store the localized package data as a separate record.
-        $languagedata = array();
-        foreach ($this->languages as $language) {
-            $languagedata[] = [
-                "packageid" => $packageid,
-                "language" => $language,
-                "name" => $this->get_localized_property($this->name, [$language]),
-                "description" => $this->get_localized_property($this->description, [$language])
-            ];
+        ]);
+
+        if ($this->languages) {
+            // For each language store the localized package data as a separate record.
+            $languagedata = [];
+            foreach ($this->languages as $language) {
+                $languagedata[] = [
+                    "packageid" => $packageid,
+                    "language" => $language,
+                    "name" => $this->get_localized_property($this->name, [$language]),
+                    "description" => $this->get_localized_property($this->description, [$language])
```

Comment:
Theoretisch gibt es für die beiden Properties die Convenience-Methoden `get_localized_name` und `get_localized_description`. Da der "Convenience-Faktor" relativ gering ist, könnte man auch überlegen, diese Methoden zu entfernen.

```suggestion
                    "name" => $this->get_localized_name([$language]),
                    "description" => $this->get_localized_description([$language])
```